scripts/check-report-data-truth-closure.mjs

```
@@ -0,0 +1,32 @@
+for (const marker of ['get_sales_export_rows', 'get_purchase_export_rows', 'get_inventory_export_rows', 'get_receivables_export_rows', 'EXPORT_TOO_LARGE', 'TENANT_CONTEXT_MISMATCH']) if (!exportSql.includes(marker)) throw new Error(`export contract missing: ${marker}`);
+if (renderer.includes('input.rows.slice(0, 42)')) throw new Error('PDF renderer still truncates exports to the first 42 rows');
+if (!renderer.includes('const pageLines = 42')) throw new Error('PDF renderer page-boundary contract missing');
+
+const invoices = Array.from({ length: 21 }, () => ({ total: 100, status: 'confirmed' }));
+const canonical = invoices.reduce((sum, row) => sum + row.total, 0);
+const firstPage = invoices.slice(0, 20).reduce((sum, row) => sum + row.total, 0);
+if (canonical !== 2100 || firstPage !== 2000) throw new Error('21-row pagination regression fixture invalid');
+if (canonical === firstPage) throw new Error('pagination-derived business total regression not detected');
+
+const unknownInventory = [{ quantity: 10, unit_cost: 2 }, { quantity: null, unit_cost: 5 }, { quantity: 3, unit_cost: null }];
+if (unknownInventory.some(row => row.quantity == null || row.unit_cost == null) && 0 === 0) {
```

Biome (2.5.7)

[error] This comparison uses the same expression on both sides.

(lint/suspicious/noSelfCompare)